### PR TITLE
docs(compute-samples): added samples for moving VM instances

### PR DESCRIPTION
**Pull request by Sita04**
Added samples to move instances across zones (https://cloud.google.com/compute/docs/instances/moving-instance-across-zones)

Apologies for the PR size.

### CODE REVIEW

**Comment by Shabirmean** [comment]
Thank you very much for the samples @Sita04 , these looks great 😃 

Just a few thoughts:
- I see that there are multiple different actions that are part of this PR. I feel like it will make it easy on the reviewers if this was broken into multiple PRs.
   - Get instance
   - ListDisks, CreateEmptyDisk, DeleteDisk 
   - CreateWithDisks, CreateDiskFromImage, CreateDiskFromSnapshot
   - ListSnapshots, CreateSnapshot, DeleteSnapshot
   - SetDiskAutodelete

- Breaking up the PR will also ensure that important details of the PR are not overlooked by the reviewers

- Also, was the expectation to write a specific sample to showcase instance migration to a new zone? I looked around and couldn't exactly find a specific sample that shows this action.

Let me know what you think.

**Comment by Sita04** [comment]
Thanks for the review, @Shabirmean :)

Agree with the PR splitting strategy. Would be easier for the reviewers. I will be more careful the next time. Thanks for pointing it out.

> the expectation to write a specific sample to showcase instance migration to a new zone?

Yes, these samples represent a collection that could be used in the doc under moving VM instances page. They do not refer to a sample on how to move instances.
Please refer to this bug for more info: b/202504009

File: compute/cloud-client/src/main/java/compute/disks/CreateSnapshot.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    String region = "your-disk-region";
+
+    // The Cloud Storage multi-region or the Cloud Storage region where you
+    // want to store your snapshot.
+    // You can specify only one storage location. Available locations:
+    // https://cloud.google.com/storage/docs/locations#available-locations
+    String location = "europe-central2";
+
+    // project ID or project number of the Cloud project that
+    // hosts the disk you want to snapshot. If not provided, will look for
```
Comment (savijatv) [inline compute/cloud-client/src/main/java/compute/disks/CreateSnapshot.java:61]: Please review and change this text accordingly.

If you do not provide an ID or number for the source disk project, a disk from the project specified by the `projectId` will be taken for creating the snapshot.

File: compute/cloud-client/src/main/java/compute/disks/DeleteDisk.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+public class DeleteDisk {
+
+  public static void main(String[] args)
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    // TODO(developer): Replace these variables before running the sample.
+
+    // Project ID or project number of the Cloud project you want to use.
+    String projectId = "YOUR_PROJECT_ID";
+
+    // Name of the zone in which is the disk you want to delete.
```
Comment (savijatv) [inline compute/cloud-client/src/main/java/compute/disks/DeleteDisk.java:36]: The zone from where you want to delete the disk.

File: compute/cloud-client/src/main/java/compute/disks/ListDisks.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+public class ListDisks {
+
+  public static void main(String[] args)
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    // TODO(developer): Replace these variables before running the sample.
+
+    // Project ID or project number of the Cloud project you want to use.
+    String projectId = "YOUR_PROJECT_ID";
+
+    // Name of the zone in which to list the disks.
```
Comment (savijatv) [inline compute/cloud-client/src/main/java/compute/disks/ListDisks.java:35]: The zone where the disks are located.

File: compute/cloud-client/src/main/java/compute/disks/SetDiskAutodelete.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+public class SetDiskAutodelete {
+
+  public static void main(String[] args)
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    // TODO(developer): Replace these variables before running the sample.
+
+    // Project ID or project number of the Cloud project you want to use.
+    String projectId = "YOUR_PROJECT_ID";
+
+    // Name of the zone in which is the disk you want to modify.
```
Comment (savijatv) [inline compute/cloud-client/src/main/java/compute/disks/SetDiskAutodelete.java:37]: The zone of the disk that you want to modify.

File: compute/cloud-client/src/main/java/compute/disks/SetDiskAutodelete.java [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+    String zone = "europe-central2-b";
+
+    // Name of the instance the disk is attached to.
+    String instanceName = "YOUR_INSTANCE_NAME";
+
+    // The name of the disk which flag you want to modify.
```
Comment (savijatv) [inline compute/cloud-client/src/main/java/compute/disks/SetDiskAutodelete.java:43]: The name of the disk that you want to modify? or The name of the disk for which you want to modify the autodelete flag?

File: compute/cloud-client/src/main/java/compute/disks/CreateSnapshot.java (hunk 1 above)
Comment (Sita04) [reply compute/cloud-client/src/main/java/compute/disks/CreateSnapshot.java]: Done, modified it to sound better. PTAL.

File: compute/cloud-client/src/main/java/compute/CreateInstanceWithExistingDisks.java [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+
+    // Name of the zone to create the instance in. For example: "us-west3-b"
+    String zone = "europe-central2-b";
+
+    // Name of the new virtual machine (VM) instance.
+    String instanceName = "YOUR_INSTANCE_NAME";
+
+    // Array of disk names to be attached to the new virtual machine.
+    // First disk in this list will be used as the boot disk.
+    List<String> diskNames = new ArrayList<>();
```
Comment (Shabirmean) [inline compute/cloud-client/src/main/java/compute/CreateInstanceWithExistingDisks.java:51]: Should we have a list of names here?  I think we should add some examples here for the user to change. If not it looks as if an empty list is okay.

```suggestion
    List<String> diskNames = List.of("your-boot-disk", "another-disk1", "another-disk2");
```

File: compute/cloud-client/src/main/java/compute/CreateInstanceWithExistingDisks.java [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+    try (InstancesClient instancesClient = InstancesClient.create();
+        DisksClient disksClient = DisksClient.create()) {
+
+      if (diskNames.size() == 0) {
+        throw new Error("At least one disk should be provided");
+      }
+
+      // Create the list of attached disks to be used in instance creation.
+      List<AttachedDisk> attachedDisks = new ArrayList<>();
+      Disk disk = null;
```
Comment (Shabirmean) [inline compute/cloud-client/src/main/java/compute/CreateInstanceWithExistingDisks.java:75]: Why define this outside? Can we move it into the for loop?

File: compute/cloud-client/src/main/java/compute/CreateInstanceWithExistingDisks.java [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+        attachedDisks.add(
+            AttachedDisk.newBuilder()
+                .setSource(disk.getSelfLink())
+                .build()
+        );
+      }
+
+      // Make the first disk in the list as the boot disk.
+      attachedDisks.set(0,
+          AttachedDisk.newBuilder(attachedDisks.get(0)).setBoot(true).build());
```
Comment (Shabirmean) [inline compute/cloud-client/src/main/java/compute/CreateInstanceWithExistingDisks.java:87]:
```suggestion
      for (int i = 0; i < diskNames.size(); i++) {
        String diskName = diskNames.get(i);
        Disk disk = disksClient.get(projectId, zone, diskName);
        AttachedDisk attDisk = null;

        if (i == 0) {
            // Set the first disk in the list as boot disk.
            attDisk = AttachedDisk.newBuilder().setSource(disk.getSelfLink()).setBoot(true).build();
        } else {
            attDisk = AttachedDisk.newBuilder().setSource(disk.getSelfLink()).build();
        }
        attachedDisks.add(attDisk);
      }
```

File: compute/cloud-client/src/main/java/compute/disks/CreateSnapshot.java [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+    String zone = "europe-central2-b";
+
+    // The region of the source disk from which you create the snapshot (for regional disks).
+    String region = "your-disk-region";
+
+    // The Cloud Storage multi-region or the Cloud Storage region where you
+    // want to store your snapshot.
+    // You can specify only one storage location. Available locations:
+    // https://cloud.google.com/storage/docs/locations#available-locations
+    String location = "europe-central2";
```
Comment (Shabirmean) [inline compute/cloud-client/src/main/java/compute/disks/CreateSnapshot.java:58]: Are the regions and locations the same? As in are they from the same list? if so can we use one variable for both here?

File: compute/cloud-client/src/main/java/compute/disks/ListDisks.java [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+      ListDisksRequest listDisksRequest = ListDisksRequest.newBuilder()
+          .setProject(projectId)
+          .setZone(zone)
+          .setFilter(filter)
+          .build();
+
+      for (Disk disk : disksClient.list(listDisksRequest).iterateAll()) {
+        System.out.println(disk.getName());
+      }
+      System.out.println("Listed all disks.");
```
Comment (Shabirmean) [inline compute/cloud-client/src/main/java/compute/disks/ListDisks.java:65]: Can we move this to be printed before the for loop and say `List of disks:`

File: compute/cloud-client/src/main/java/compute/disks/ListSnapshots.java [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+      // Create the List Snapshot request.
+      ListSnapshotsRequest listSnapshotsRequest = ListSnapshotsRequest.newBuilder()
+          .setProject(projectId)
+          .setFilter(filter)
+          .build();
+
+      for (Snapshot snapshot : snapshotsClient.list(listSnapshotsRequest).iterateAll()) {
+        System.out.println(snapshot.getName());
+      }
+      System.out.println("Listed all snapshots");
```
Comment (Shabirmean) [inline compute/cloud-client/src/main/java/compute/disks/ListSnapshots.java:57]: Same as ListDisk, can we move it before the for loop and say `List of snapshots:`

**Review by savijatv — APPROVED** [review]

File: compute/cloud-client/src/main/java/compute/CreateInstanceWithExistingDisks.java (hunk 6 above)
Comment (Sita04) [reply compute/cloud-client/src/main/java/compute/CreateInstanceWithExistingDisks.java:51]: That's a great suggestion. I've added it. Thanks.

File: compute/cloud-client/src/main/java/compute/disks/CreateSnapshot.java (hunk 9 above)
Comment (Sita04) [reply compute/cloud-client/src/main/java/compute/disks/CreateSnapshot.java:58]: They are different in that locations might have dual or multiple regions. The location list may also vary with time and need not always be the same as the region list. 
Hence, I would prefer to keep it as two separate variables, if that makes sense?

**Comment by Sita04** [comment]
@Shabirmean Thanks for the review. I've incorporated your suggestions. PTAL.

File: compute/cloud-client/src/main/java/compute/disks/CreateSnapshot.java (hunk 9 above)
Comment (Shabirmean) [reply compute/cloud-client/src/main/java/compute/disks/CreateSnapshot.java:58]: Sounds good to me

**Review by Shabirmean — APPROVED** [review]
LGTM! Thank you for taking in all the suggestions!

**Pull request merged**